ICP-2850 Setting color temperature causes offline device to come online

greens opened the pull request:
The event generated by setting the color name is causing devices to be marked online.

greens committed:
ICP-2850 Setting color temperature causes offline device to come online

The event generated by setting the color name is causing devices to be marked online.

greens commented:
So a couple of questions about this.

This fix is an option, where we won't set the color temperature unless we get an update from the device.

We could also change the logic to only change the color temperature if the device is online, but that may have side effects.

Or we could get the device health team to do extra checking around which events cause devices to come back online

@tpmanley Would this change in the DTH also fix the issue for local execution lights?

greens commented on devicetypes/smartthings/zll-rgbw-bulb.src/zll-rgbw-bulb.groovy:
```diff
@@ -153,6 +153,7 @@ def installed() {
 }
 
 def setColorTemperature(value) {
+    // TODO: Use this method for all zigbee white lights, but also change logic in appengine-zigbee to round
```
Noticed this is a copy of the truncating logic I just fixed in the DTHs. Only used in this one place, but if we fix it in appengine it could be used for all of these.

tpmanley replied:
Since `zigbee.setColorTemperature` truncates instead of rounds do you want to change it to round here and then update all the DTHs to call `zigbee.setColorTemperature()` once it's fixed there?

greens replied:
Yeah. Seems like the right thing to do.

tpmanley commented:
I believe you can remove `command "setGenericName"` from all the DTHs since I don't think any SmartApps can or should be calling it.

tpmanley commented:
Consider adding the `setGenericName` stuff to zll-rgbw-bulb.groovy to make it consistent with the other RGBW/TW DTHs.

tpmanley commented on devicetypes/smartthings/zll-white-color-temperature-bulb.src/zll-white-color-temperature-bulb.groovy:
```diff
@@ -79,6 +79,9 @@ def parse(String description) {
     log.debug "description is $description"
     def event = zigbee.getEvent(description)
     if (event) {
+        if (event.name=="colorTemperature") {
```
nit: spaces around `==`

greens committed:
ICP-2850 Setting color temperature causes offline device to come online

Incorporating review suggestions.

greens committed:
ICP-2850 Setting color temperature causes offline device to come online

Incorporating review suggestions.

greens commented:
@tpmanley Incorporated some of your comments. I didn't add the color temperature attribute to the rgbw bulb since my guess is that it was left off due to confusion it might cause with the color vs color temp.

workingmonk commented:
setGenericName was added as a command because the first iteration had a tile to quickly switch between the value color temp values. over time that tile action was taken out, but the command persisted

tpmanley commented:
@greens unless I'm missing something it's inconsistent between the Zigbee RGBW Bulb and the ZLL RGBW Bulb DTH and I don't think there is any reason it shouldn't be the same

greens committed:
ICP-2850 Setting color temp causes offline bulbs to be marked online

Changes the zll rgbw bulb to match the zigbee rgbw bulb in terms of how it handles color temperature.

greens commented:
@tpmanley Done.

tpmanley reviewed (approved):
Looks good, just squash the commits before the merge

berkens commented:
I noticed that the color temperature can be changed by the slider from 2700K to 6500K.

However 2700K is not always the lower limit of the bulb / controller. 

It would be nice if the lower limit was 2000K. 

I returned an Osram bulb because it was not so warm white. 

Later on I experienced that a led strip controller with the same DTH the warm white leds were illuminated, but the cool white leds were also still on (dim). 

I changed the DTH for my led controller so the color temperature range is now 2000K - 6500K. The cool white leds are now off at 2000K.
